src/main/java/ua/softserveinc/tc/util/BookingsCharacteristics.java

```diff
@@ -360,4 +360,57 @@ private BookingsCharacteristics(Builder builder) {
         users = builder.users;
         bookingsStates = builder.bookingsStates;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+
+            return true;
+        }
+        if (!(o instanceof BookingsCharacteristics)) {
+
+            return false;
+        }
+
+        BookingsCharacteristics characteristics = (BookingsCharacteristics) o;
+
```

Comment:
It is wrong thoughts. Because this situation is impossible. The point is in insatnceof operator. According to §15.20.2. of JLS, if the RelationalExpression operand is null, then result of this operation is false, so no casting will be processed and no NullPointerException is produces. And result will be false.

i.e. -> null instanceof Object  -> is false!

https://docs.oracle.com/javase/specs/jls/se8/html/jls-15.html#jls-15.20.2